Source/JavaScriptCore/runtime/DeferredWorkTimer.cpp

```diff
@@ -240,6 +244,31 @@ void DeferredWorkTimer::cancelPendingWorkSafe(JSGlobalObject* globalObject)
         setTimeUntilFire(0_s);
 }
 
+void DeferredWorkTimer::cancelPendingWork(VM& vm)
+{
+    ASSERT(vm.heap.isInPhase(CollectorPhase::End));
+    Locker locker { m_taskLock };
+
+    auto isValid = [&](auto& ticket) {
+        for (JSCell* dependency : ticket->dependencies()) {
+            if (!vm.heap.isMarked(dependency))
+                return false;
+        }
```

Comment:
```
If ticket->target()->globalObject() is marked then these will all be marked because it visits the dependencies.
```
Yes, that should be true. But we still need to check the target. This is because if the target is dead its global object may not be dead.

Reply:
The global object marks everything in `m_dependencies` though. So target can't be dead.

Reply:
Updated.

Reply:
Updated.
```
    auto isValid = [&](auto& ticket) {
        bool isTargetGlobalObjectLive = vm.heap.isMarked(ticket->target()->globalObject());
#if ASSERT_ENABLED
        for (JSCell* dependency : ticket->dependencies())
            ASSERT(vm.heap.isMarked(dependency) == isTargetGlobalObjectLive);
#endif
        return isTargetGlobalObjectLive && vm.heap.isMarked(ticket->scriptExecutionOwner());
    };
```